films_imdb/films_functions_spark/function_films/function_imdb3.py

```diff
@@ -0,0 +1,156 @@
+
+
+# best films sorted by genre and ranked by rating for the period from 1990 to 2000
+def best_films_by_genres_1990th(df):
+
+    df = df.filter((f.col("titleType") == "movie") & (f.col("numVotes") > 100000)
+                   & ((f.col("startYear") >= 1990) & (f.col("startYear") <= 1999)))\
+        .withColumn("yearRange", f.lit("1990-2000"))
+
+    df = df.withColumn("genres", split_genre)\
+        .withColumn("row_number", f.row_number().over(window))\
+        .filter((f.col("row_number") < 11))
+
+    return df.orderBy("genres", "averageRating", "row_number", ascending=False)\
+        .select("tconst", "primaryTitle", "startYear", "genres", "averageRating", "numVotes", "yearRange")
+
+
+# best films sorted by genre and ranked by rating for the period from 1980 to 1990
+def best_films_by_genres_1980th(df):
+
+    df = df.filter((f.col("titleType") == "movie") & (f.col("numVotes") > 100000)
+                   & ((f.col("startYear") >= 1980) & (f.col("startYear") <= 1989)))\
+        .withColumn("yearRange", f.lit("1980-1990"))
+
+    df = df.withColumn("genres", split_genre)\
+        .withColumn("row_number", f.row_number().over(window))\
+        .filter((f.col("row_number") < 11))
+
+    return df.orderBy("genres", "averageRating", "row_number", ascending=False)\
+        .select("tconst", "primaryTitle", "startYear", "genres", "averageRating", "numVotes", "yearRange")
+
+
+# best films sorted by genre and ranked by rating for the period from 1970 to 1980
+def best_films_by_genres_1970th(df):
+
+    df = df.filter((f.col("titleType") == "movie") & (f.col("numVotes") > 100000)
+                   & ((f.col("startYear") >= 1970) & (f.col("startYear") <= 1979)))\
+        .withColumn("yearRange", f.lit("1970-1980"))
+
+    df = df.withColumn("genres", split_genre)\
+        .withColumn("row_number", f.row_number().over(window))\
+        .filter((f.col("row_number") < 11))
+
+    return df.orderBy("genres", "averageRating", "row_number", ascending=False)\
+        .select("tconst", "primaryTitle", "startYear", "genres", "averageRating", "numVotes", "yearRange")
+
+
+# best films sorted by genre and ranked by rating for the period from 1960 to 1970
+def best_films_by_genres_1960th(df):
+
+    df = df.filter((f.col("titleType") == "movie") & (f.col("numVotes") > 100000)
+                   & ((f.col("startYear") >= 1960) & (f.col("startYear") <= 1969)))\
+        .withColumn("yearRange", f.lit("1960-1970"))
+
+    df = df.withColumn("genres", split_genre)\
+        .withColumn("row_number", f.row_number().over(window))\
+        .filter((f.col("row_number") < 11))
+
+    return df.orderBy("genres", "averageRating", "row_number", ascending=False)\
+        .select("tconst", "primaryTitle", "startYear", "genres", "averageRating", "numVotes", "yearRange")
+
+
+#  best films sorted by genre and ranked by rating for the period from 1950 to 1960
+def best_films_by_genres_1950th(df):
```

Comment:
These  functions are the same, it should be one function with arguments in order not to duplicate code